### DISCUSSION TITLE
Add sorting by o2m aggregation

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#6286] [Archive]
Hi, it seems impossible to:

1. `sort` by count. For instance, if I have a "Library" collection with a m2m field that links to a "Book" collection, how do I query Directus to obtain a list of Libraries by their Book count?
2. `sort` by foreign key fields. For instance, If I have "Post" collection and each post has a `user_created` field, I'd expend `?sort[]=-user_created.first_name` to work, but I receive a `Unknown column 'posts.user_created.first_name' in 'order clause'",` error.

Thanks for any help you'd provide.

**Comment by rijkvanzanten** [comment]
Number 2 is a duplicate of https://github.com/directus/directus/discussions/4502
